pkg/graveler/ref/address_token_iterator.go

```diff
@@ -0,0 +1,98 @@
+package ref
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+// AddressTokenIterator Iterates over repository's token addresses
+type AddressTokenIterator struct {
+	ctx           context.Context
+	it            kv.MessageIterator
+	err           error
+	value         *graveler.AddressData
+	repoPartition string
+	store         kv.Store
+	closed        bool
+}
+
+func NewAddressTokenIterator(ctx context.Context, store *kv.StoreMessage, repo *graveler.RepositoryRecord) (*AddressTokenIterator, error) {
+	repoPartition := graveler.RepoPartition(repo)
+	it, err := kv.NewPrimaryIterator(ctx, store.Store, (&graveler.AddressData{}).ProtoReflect().Type(),
+		repoPartition, []byte(graveler.AddressPath("")), kv.IteratorOptionsFrom([]byte("")))
+	if err != nil {
+		return nil, err
+	}
+
+	return &AddressTokenIterator{
+		ctx:           ctx,
+		store:         store.Store,
+		it:            it,
+		repoPartition: repoPartition,
+		value:         nil,
+		err:           nil,
+	}, nil
+}
+
+func (i *AddressTokenIterator) Next() bool {
+	if i.Err() != nil || i.closed {
+		return false
+	}
+	if !i.it.Next() {
+		i.value = nil
+		return false
+	}
+	e := i.it.Entry()
+	if e == nil {
+		i.err = graveler.ErrReadingFromStore
+		return false
+	}
+	token, ok := e.Value.(*graveler.AddressData)
```

Comment:
This can't really happen so I recommend removing it altogether

Reply:
why it can't happen? if we saved under this prefix not address data